analytics/terraform/spark-eks-ipv6/vpc.tf

```
@@ -0,0 +1,36 @@
+# VPC with IPv6 capability
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 5.0"
+
+  name = local.name
+  cidr = local.vpc_cidr
+
+  azs             = local.azs
+  private_subnets = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 4, k)]
+  public_subnets  = [for k, v in local.azs : cidrsubnet(local.vpc_cidr, 8, k + 48)]
+
+  enable_nat_gateway     = true
+  single_nat_gateway     = true
+  enable_ipv6            = true
+  create_egress_only_igw = true
+
+  # IPv6-only subnets config
+  public_subnet_ipv6_prefixes                    = [0, 1, 2]
+  public_subnet_assign_ipv6_address_on_creation  = true
+  private_subnet_ipv6_prefixes                   = [3, 4, 5]
+  private_subnet_assign_ipv6_address_on_creation = true
+
```

The IPv6 support in the terraform module seems to set DNS64=true on the subnets that are created. This can cause unexpected traffic routing for ipv4 endpoints as described here https://github.com/aws/containers-roadmap/issues/2352 

Since EKS will handle 6->4 SNAT at the node we kinda want the DNS to resolve to the actual IP, whether thats IPv4 or v6. I think you can use a flag like these to disable it for the subnets in the example
```suggestion
public_subnet_enable_dns64 = false
private_subnet_enable_dns64 = false
```

---

Disabled DNS64